js/buttons.js

```diff
@@ -56,10 +54,12 @@ function addButtons() {
       return (d + "-button-text");
     })
     .style('font-weight', 'thick')
-    .attr('text-anchor', 'end');
+    .attr('text-anchor', 'end')
+    .each( function(d) { this.__data__ = {category: d, wateruse: NaN}; });
```

Comment:
Everything we've appended to `buttons` has automatically inherited the `__data__` attribute of its parent button, which is one element of the `categories` vector, e.g., `'total'`. But for us to be able to merge the `category-amount` text elements with new data when we switch views, we need both the category and the wateruse value to be available, so we need a structure that can match the actual data we'll be passing in later. This line creates that structure by modifying the `__data__` attribute of each `text.category-amount` element.